apps/etterlatte-saksbehandling-ui/client/src/components/behandling/revurderingsoversikt/SoeskenjusteringUndertype.tsx

```diff
@@ -0,0 +1,91 @@
+import { IDetaljertBehandling } from '~shared/types/IDetaljertBehandling'
+import {
+  BarnepensjonSoeskenjusteringGrunn,
+  RevurderingInfo,
+  SOESKENJUSTERING_GRUNNER,
+  SoeskenjusteringInfo,
+  tekstSoeskenjustering,
+} from '~shared/types/RevurderingInfo'
+import { FormEvent, useState } from 'react'
+import { BodyShort, Button, Heading, Select } from '@navikt/ds-react'
+import { hentBehandlesFraStatus } from '~components/behandling/felles/utils'
+import { isPending, useApiCall } from '~shared/hooks/useApiCall'
+import { lagreRevurderingInfo } from '~shared/api/revurdering'
+import { ApiErrorAlert } from '~ErrorBoundary'
+import { oppdaterRevurderingInfo } from '~store/reducers/BehandlingReducer'
+import styled from 'styled-components'
+
+function hentUndertypeFraBehandling(behandling?: IDetaljertBehandling): SoeskenjusteringInfo | null {
+  const revurderinginfo = behandling?.revurderinginfo
+  if (revurderinginfo?.type == 'SOESKENJUSTERING') {
+    return revurderinginfo
+  } else {
+    return null
+  }
+}
+
+export const SoeskenjusteringUndertype = (props: { behandling: IDetaljertBehandling }) => {
+  const { behandling } = props
+  const soeskenjusteringInfo = hentUndertypeFraBehandling(behandling)
+  const [valgtSoeskenjustering, setValgtSoeskenjustering] = useState<BarnepensjonSoeskenjusteringGrunn | undefined>(
+    soeskenjusteringInfo?.grunnForSoeskenjustering
+  )
+  const [feilmelding, setFeilmelding] = useState<string | undefined>(undefined)
+  const [lagrestatus, lagre] = useApiCall(lagreRevurderingInfo)
+  const redigerbar = hentBehandlesFraStatus(behandling.status)
+  const handlesubmit = (e: FormEvent) => {
+    e.stopPropagation()
+    setFeilmelding(undefined)
+    if (valgtSoeskenjustering === undefined) {
+      setFeilmelding('Du må velge hvorfor du gjennomfører en søskenjustering')
```

Comment:
Holder det ikke å ha denne valideringen på feltet med required attributtet også vil den slå til ved onsubmit og?

Reply:
siden dette er en useState-verdi vil ikke required gi info til typescript om at den *ikke* er undefined, så tenkte det letteste var å løse det felles her. Men åpen for innspill :)

Reply:
Sant nok det! Det er kanskje bedre enn å sjekke `form.checkValidity()` også ha en custom med `setCustomValidity` ? :D